### ISSUE TITLE
Using LMS-ESP32 with Pybricks

### ISSUE THREAD

**Issue by aivarannamaa** [antonvh/LMS-uart-esp#14]
I'm interested in creating a custom input device for LEGO Technic Hub and I'm currently investigating the alternatives.

It looks like this hub is supported by Pybricks and from the documentation it looks like there is API for communicating with generic 3rd party devices (https://docs.pybricks.com/en/stable/iodevices/index.html).

What do you think, would it be possible to communicate with LMS-ESP32 using the Pybricks firmware?

**Comment by antonvh** [comment]
Yes it is possible. But be aware:
- Pybricks does not support UART (yet)
- The way to communicate is through the LEGO Powered Up/LPF2  Protocol.
- We have written a library to emulate that protocol.
https://github.com/antonvh/LMS-uart-esp/tree/main/Libraries/LPF2 <https://github.com/antonvh/LMS-uart-esp/tree/main/Libraries/LPF2>
- The cool thing is that with some hacks, the Scratch interface will also work with your sensor.

**Comment by aivarannamaa** [comment]
Thanks for this information! I'll order the board and start experimenting.

**Comment by ste7anste7an** [comment]
The LMS_esp32.py library has been updated. I could successfully read the LMS-ESP32 'sensor' from Pybricks. However, writing data to the sensor, is not correctly implemented in PyBricks. We might open an issue with PyBricks.

I installed `LMS_esp32.py` on the LMS-ESP32. Then I ran the test program `LMS_test_esp32.py`. On PyBricks I used the following code:

```
from pybricks.iodevices import PUPDevice
from pybricks.parameters import Port
esp32 = PUPDevice(Port.D)
for i in range(100):
  print(esp32.read(0))
```

In this demo I could not write to mode 0 or mode 1, however mode 2 (single float) worked:
```
esp32.write(2,(10,))
```
